### PR TITLE
Fix Data Length Procedure tests to skip IUT unsupported parameters

### PR DESCRIPTION
**Pull request by cvinayak**
Relates to: https://github.com/zephyrproject-rtos/zephyr/pull/38407

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Review by wopu-ot — APPROVED** [review]
Changes look good to me, but I would also like a second opinion from @erbr-ot before merging.

**Comment by kruithofa** [comment]
What problem is fixed by this?

**Comment by cvinayak** [comment]
@wopu-ot There is something fishy, 4 SLA tests needed the requested flag, but only for 2 MAS tests suffice. There may be more to these tests.

File: src/tests/ll_verification.py
Comment (erbr-ot) [inline src/tests/ll_verification.py:3552]: I have a hard time understanding this 'changed' abstraction. How come we need the trailing 'or (cmaxTxTime == min(txTime, 328))' part?

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@erbr-ot @wopu-ot can someone merge this please?

File: src/tests/ll_verification.py
Comment (cvinayak) [reply src/tests/ll_verification.py:3552]: That is original code, not changed in this PR.

Comment (erbr-ot) [reply src/tests/ll_verification.py:3552]: Yes, I realize this. Sorry for being unclear. 
I'm just curious about this, and I mentioned it because I wonder if this is wrong and perhaps then hides other issues re. DLE testing. Let's continue this elsewhere;-)

**Pull request merged**
